docs/ref-manual/cvl/methods.md

```diff
@@ -46,44 +55,280 @@ method_summary   ::= "ALWAYS" "(" value ")"
 
-### `DISPATCHER`
+The most commonly used option is `DISPATCHER(true)`, because in most cases the
+behavior of `DISPATCHER(false)` is equivalent to that of `AUTO`.
+`DISPATCHER(false)` only behaves differently from `AUTO` when one of the known
+contracts exhibits behavior that `AUTO` assumes is impossible (such as
+reentrancy).
 
-### `AUTO`
+(auto-summary)=
+### `AUTO` summaries
 
-### expression and block summaries
+The behavior of the `AUTO` summary depends on the type of call[^opcodes]:
 
+ * Calls to non-library `view` and `pure` methods use the `NONDET` approximation:
+   they keep all state unchanged.
 
+ * Normal calls and constructors use the `HAVOC_ECF` approximation: they are
+   assumed to change the state of external contracts arbitrarily but to leave
+   the caller's state unchanged.
+
+ * Calls to library methods are assumed to change the caller's state in an
+   arbitrary way, but are assumed to leave the state of other contracts
+   unchanged.
+
+```{todo}
+The effect of library calls on the current contract's ETH balance is unclear.
+
+The following comment from the old docs should be clarified:
+
+If the contract you are verifying relies heavily on modification of ETH
+balances, it's recommended to identify the balance-modifying functions and mark
+them `HAVOC_ALL` if necessary.
+```
+
+[^opcodes]: The behavior of `AUTO` summaries is actually determined by the EVM
+  opcode used to make the call: calls made using the `STATICCALL` opcode use
+  the `NONDET` summary, calls using `CALL` or `CREATE` opcode use the `HAVOC_ECF`
+  summary, and calls using the `DELEGATECALL` and `CALLCODE` opcodes havoc the
+  current contract only.
+  Modern Solidity versions output opcodes that are consistent with the above
+  description, but older versions behave differently.  See
+  [State Mutability](https://docs.soliditylang.org/en/v0.8.12/contracts.html#state-mutability)
+  in the Solidity manual for details.
+
+(function-summary)=
+### Function summaries
+
+Contract methods can also be summarized using CVL {doc}`functions` or
+{ref}`ghost-axioms` as approximations.  Contract calls to the summarized method
+are replaced by calls to the specified CVL functions.
+
+To use a CVL function or ghost as a summary, use a call to the function in
+place of the summary type.  You may also write `with(env e)` before the call to
+the CVL function to make the environment `e` available to the function call.
+The function call can only refer directly to the environment and variables
+defined as arguments in the summary declarations; expressions involving those
+variables are not supported.
+
+Functions used as summaries are not allowed to call contract functions.  They
+may only accept parameter types that are expressible in solidity; extended CVL
+types like `method` and `mathint` cannot be passed as arguments.
+
+### Undocumented summaries
+
+```{todo}
+Block summaries are undocumented.  Expression summaries other than function
```

Comment:
block summaries don't exist
expression summaries other than function summaries don't exist
don't let the parser fool you :P

Reply:
removed